src/sphinxfilter.cpp

```diff
@@ -951,8 +977,10 @@ static std::unique_ptr<ISphFilter> CreateFilter ( const CSphFilterSettings & tSe
 			else if ( tSettings.m_eMvaFunc==SPH_MVAFUNC_ALL )
 				return std::make_unique<Filter_StringTagsAll_c>();
 		}
-		else
+		else if ( tSettings.m_eStrCmpDir==EStrCmpDir::EQ )
 			return std::make_unique<FilterString_c> ( eCollation, tSettings.m_bHasEqualMin || tSettings.m_bHasEqualMax );
+		else
+			return std::make_unique<FilterStringCmp_c> ( eCollation, tSettings.m_bHasEqualMin || tSettings.m_bHasEqualMax, tSettings.m_eStrCmpDir );
```

Comment:
here is the bug. It definitely should be `tSettings.m_bExclude` here instead of `tSettings.m_bHasEqualMin || tSettings.m_bHasEqualMax`.
But it will break current logic, since this one is shadowed by inversion node.
So, also 'bool		CanExclude() const override { return true; }' should be added into string comparator then.